lib/CGameInfoCallback.cpp

```diff
@@ -55,7 +55,11 @@ bool CGameInfoCallback::isAllowed( int type, int id )
 	case 1:
 		return gs->map->allowedArtifact[id];
 	case 2:
-		return gs->map->allowedAbilities[id];
+		// new skills are always allowed
+		if(id >= gs->map->allowedAbilities.size())
```

Comment:
Our editor and VCMI map format will allow new skills disablement once this branch is landed so you need to support that. I 99% sure this already work for custom spells so you'll likely just need to copy-paste a bit of code used for ```allowedSpell```.

Reply:
This just need to be reverted `allowedAbilities.size()` shall equal actual number of skills. And getDefaultAllowedAbilities() should be moved to SkillHandler